src/emulate.c

```diff
@@ -1111,6 +1116,41 @@ void rv_step(void *arg)
 #endif
 }
 
+void rv_step_debug(void *arg)
+{
+    assert(arg);
+    riscv_t *rv = arg;
+
+    rv_check_interrupt(rv);
```

Comment:
How about modify here like following?
```
#if RV32_HAS(SYSTEM) && !RV32_HAS(ELF_LOADER)
    rv_check_interrupt(rv);
#endif
```

In this way, the dummy `rv_step_debug` can be removed.

Reply:
The debug mode of system simulation is not supported yet. Should we remove this statement to avoid ambiguity?

Reply:
What does "debug mode of system simulation is not supported" mean here? The debug/gdbstub mode should be supported naturally because there's no big difference between program or system emulation. I already check that.

Reply:
Would `rv32emu` support debugging with booting Linux Kernel now? I failed to run `file build/linux-image/Image` in `riscv32-unknown-elf-gdb` to launch the remote GDB since it is not the RISC-V executable.

Reply:
You can check like this
1. Open a bash shell and run `./build/rv32emu -k Image -i rootfs.cpio -g`
2. Open another shell and run `riscv32-unknown-elf-gdb -ex "target remote :1234" -ex "cont" -ex "q"`

Reply:
Oh, it works. Thank you to point it out. Can you also update the document for this? (maybe in another PR)